OpenRA.Game/Translation.cs

```diff
@@ -88,6 +88,9 @@ public string GetString(string key, IDictionary<string, object> arguments = null
 
 		public bool TryGetString(string key, out string value, IDictionary<string, object> arguments = null)
 		{
+			if (string.IsNullOrEmpty(key))
+				throw new ArgumentException("A translation key must not be null or empty.", nameof(key));
```

Comment:
A `TryGet*` method should not throw exceptions!

Reply:
I could make it return false here instead and move the throwing further up.

Reply:
Done.

Reply:
> A `TryGet*` method should not throw exceptions!

[Dictionary.TryGetValue](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2.trygetvalue?view=net-6.0#exceptions) begs to differ!

Reply:
I think the provided key being null is a valid cause for crash. You can't really try to get a null from something in the same way you can't divide by 0, and if you can then it's weird

I'm fine with either approach

Reply:
Moved it back into the TryGet method so it cannot be bypassed.